test/index.js

```diff
@@ -760,6 +761,172 @@ describe('normalizr', function () {
       }
     });
   });
+  
+  it('fails normalizing with a relationship', function () {
+    var user = {
+      id: 1,
+      relations: []
+    };
+      
+    (function () {
+      normalize(user, relationOf('userRelations'))
+    }).should.throw('Normalize does not accept a relation as a schema');
+  });
+
+  it('can normalize nested object with relationship', function() {
+    var job = new Schema("jobs");
+    var user = new Schema('users');
+    user.define({
+      relations: relationOf("userRelations", {
+        job: job
+      })
+    });
+
+    var input = {
+      id: 1,
+      name: 'Some name',
+      relations: [
+        {
+          type: 'open source',
+          job: { id: 1, title: 'maintainer' }
+        },
+        {
+          type: 'personal',
+          job: { id: 2, title: 'owner' }
+        }
+      ]
+    };
+
+    Object.freeze(input);
+
+    normalize(input, user).should.eql({
+      entities: {
+        users: {
+          1: { id: 1, name: 'Some name'}
+        },
+        userRelations: {
```

Comment:
If you're going to have relationships, it seems like they should be two-way and include `jobRelations` back to users automatically. As you have it now, this would be much better served as data on the user entity.